pandas/formats/format.py

```diff
@@ -2042,7 +2042,12 @@ def get_result_as_array(self):
         """
 
         if self.formatter is not None:
-            return np.array([self.formatter(x) for x in self.values])
+            # na_rep takes precedence over the specified formatter
+            if self.na_rep is not None:
+                return np.array([self.formatter(x) if notnull(x)
+                                 else self.na_rep for x in self.values])
```

Comment:
like this (from internals.py)


    def to_native_types(self, slicer=None, na_rep=None, quoting=None,
                        **kwargs):
        """ convert to our native types format, slicing if desired """

        values = self.values
        if slicer is not None:
            values = values[:, slicer]
        mask = isnull(values)

        rvalues = np.empty(values.shape, dtype=object)
        if na_rep is None:
            na_rep = 'NaT'
        rvalues[mask] = na_rep
        imask = (~mask).ravel()